src/Package/Impl/Shell/VsAppShell.cs

```diff
@@ -70,7 +70,13 @@ public sealed class VsAppShell : IApplicationShell, IMainThread, IIdleTimeServic
             }
         }
 
+        public static void Terminate() {
+            _instance?.Dispose();
+        }
+
         private void Initialize() {
+            AppConstants.Initialize();
```

Comment:
MEF allows export an instance not through its ctor, but through a property of another object. We use it in `RProjectLoadHooks`. In this case you can explicitly instantiate `AppConstants` and don't have to introduce `Initialize` method in the interface.